library/pk.c

```diff
@@ -232,6 +232,126 @@ int mbedtls_pk_can_do( const mbedtls_pk_context *ctx, mbedtls_pk_type_t type )
     return( ctx->pk_info->can_do( type ) );
 }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+/*
+ * Tell if a PK can do the operations of the given PSA algorithm
+ */
+int mbedtls_pk_can_do_ext( const mbedtls_pk_context *ctx, psa_algorithm_t alg,
+                           psa_key_usage_t usage )
+{
+    psa_key_usage_t key_usage;
+
+    /* A context with null pk_info is not set up yet and can't do anything.
+     * For backward compatibility, also accept NULL instead of a context
```

Comment:
Probably should document behaviour for `ctx == NULL` in pk.h

Reply:
It's already documented:
```
 * \return          0 if the context cannot do the operations on the given
 *                  type. This is always the case for a context that has
 *                  been initialized but not set up, or that has been
 *                  cleared with mbedtls_pk_free().
```

Reply:
Oh I see it's for the `ctx == NULL` specifically, I copied this code from `mbedtls_pk_can_do()` to keep compatibility with it, and it's not documented either.

Reply:
I'm inclined not to document this. We're allowed to be more robust/accepting than documented, but if we start documenting it then it becomes a promise we have to keep forever (that is, at least until 4.0). In this case, there's no uniformity in the PK module: for example, right now calling `pk_can_do( NULL, ... )` is allowed while `pk_verify( NULL, ... )` isn't. Hopefully we'll make this uniform some day, and I'd like to keep the liberty for doing it in the direction that `NULL` isn't allowed anywhere (allowing NULL is not sufficiently useful to justify the code size cost of the checks IMO).